keras_eval/eval.py

```diff
@@ -145,6 +149,32 @@ def evaluate(self, data_dir=None, top_k=1, filter_indices=None, confusion_matrix
 
         return self.probabilities, self.labels
 
+    def compute_inference_probabilities(self, concept_dictionary, probabilities):
+        '''
+        Sets probabilities if model concepts are to be combined in test
+
+        Args:
+            concept_dictionary: It is the dictionary which contains all the granular concepts and the mapping with the groups.
+            probabilities: These are computed granular probabilities
+
+        '''
+
+        if utils.compare_group_test_concepts(self.concept_labels, concept_dictionary) and utils.check_concept_unique(concept_dictionary):
+            for concept in concept_dictionary:
+
+                if concept['group'] in self.group_id_dict.keys():
+                    self.group_id_dict[concept['group']].append(concept['class_index'])
+                else:
+                    self.group_id_dict[concept['group']] = [concept['class_index']]
+
+            self.combined_probs = [[[0.0, ] * len(self.concept_labels)] * len(probabilities[0])]
```

Comment:
If we rename the function to `compute_inference_probabilities`, may we rename `combined_probs` as `inferred_probs` to be coherent?